tests/integ/scala/test_function_suite.py

```diff
@@ -2419,110 +2445,122 @@ def test_as_time(session):
 
 
 def test_as_timestamp_all(session):
-    Utils.check_answer(
-        TestData.variant1(session).select(
-            as_timestamp_ntz(col("timestamp_ntz1")),
-            as_timestamp_ntz(col("timestamp_tz1")),
-            as_timestamp_ntz(col("timestamp_ltz1")),
-        ),
-        [
-            Row(
-                datetime.strptime("2017-02-24 12:00:00.456", "%Y-%m-%d %H:%M:%S.%f"),
-                None,
-                None,
+    # not using America/Los_Angeles because pytz assign -7:53 timezone offset to it
+    pst_tz = pytz.timezone("Etc/GMT+8")
+    try:
+        if not IS_IN_STORED_PROC:
+            session.sql('alter session set timezone="America/Los_Angeles"').collect()
```

Comment:
It might not worth the effort as we only have two use cases, but we might be able to extract this `alter session` try-finally into a wrapper:

```
def some_wrapper(func):
  try:
    alter session
    func()
  finally:
    alter session
```